Move Galaxy tools to the eBCSgen repository

mopichalova opened the pull request:
Moved the top-level API functions from the [galaxytools repository](https://github.com/sybila/galaxytools/tree/main/tools/ebcsgen) to `/api` directory in this repository, the eBCSgen repository.
No modifications were made to the scripts or the original repository.

Close #115

Commit:
galaxy tools scripts for ebcsgen relocated

xtrojak reviewed (changes requested):
We don't want to move Galaxy tools here - only the top-level API functions (`.py` files). All the XML files, test data, or shed file don't belong here. And also it does not belong to Galaxy subdirectory (this should be actually removed one day when all "local" issues are resolved - e.g. properly sharing visualisations).

Commit:
removed all xml files and test data as per the feedback

Commit:
moved the top-level API functions to  directory in the root of this repository

mopichalova commented:
> We don't want to move Galaxy tools here - only the top-level API functions (`.py` files). All the XML files, test data, or shed file don't belong here. And also it does not belong to Galaxy subdirectory (this should be actually removed one day when all "local" issues are resolved - e.g. properly sharing visualisations).


Thank you for your feedback regarding the structure of the repository. I have made the necessary changes as per your suggestions. Moved the top-level API functions to directory in the root of this repository (commit 5eca9c2403887be54b727d88d0a5f4038a45f56f) and removed all xml files, test data and the Galaxy subdirectory, ensuring that only the relevant Python files remain (commit 5eca9c2403887be54b727d88d0a5f4038a45f56f).

xtrojak commented:
This is a good start, but I would improve it in following ways:

1. Rename the `api` folder to `eBCSgen/API` - we want it to be inside the Python package to make it available for imports
2. Define the argument parsing using `argparse` inside `if __name__ == '__main__':` section to allow calling it from the command line, but define the top level function separately - [a random example demonstrating this](https://github.com/grp-bork/formHTR/blob/main/select_ROIs.py).
3. (optional) Create `__init__.py` file in the `API` folder alongside with all the top-level scripts, importing all top-level functions in one place. This can simplify importing the functions when using the eBCSgen library. [Another random example](https://github.com/SmartBioTech/DeviceControl/blob/master/app/workspace/__init__.py).
4. Update readthedocs documentation - maybe there is some [argparse integration](https://sphinx-argparse.readthedocs.io/en/latest/), you have to check.

Commit:
move api to eBCSgen/API

Commit:
refactor API structure: encapsulated argument parsing in dedicated functions and updated __init__.py for simplified imports

xtrojak commented on eBCSgen/API/__init__.py:
```diff
@@ -0,0 +1,7 @@
+from .ebcsgen_ctl_model_checking import ctl_model_checking
+from .ebcsgen_generate_ts import generate_ts
+from .ebcsgen_pctl_model_checking import pctl_model_checking
+from .ebcsgen_pctl_parameter_synthesis import pctl_parameter_synthesis
+from .ebcsgen_sbml_export import sbml_export
+from .ebcsgen_simulate import simulate
+from .ebcsgen_static_analysis import static_analysis
```
```suggestion
```

I don't think these imports are necessary. Is there a specific reason for including them?

xtrojak reviewed (approved):
This looks good now! 

Not sure if imports in `__init__.py` are needed (it can be just empty), but that's a detail. 

We should note somewhere how to actually run the scripts from command line. Also the whole API should be included in the `readthedocs` documentation. You can do it now, or later (documented in #119).
